src/cli.rs

```diff
@@ -1,49 +1,139 @@
-use clap::*;
-use std::ffi::OsString;
+use crate::error::PnError;
+use crate::{commands, commands::CommandTrait, error::MainError};
+use clap::{Parser, Subcommand};
+use itertools::Itertools;
+use pipe_trait::Pipe;
+use std::{env, ffi::OsString, num::NonZeroI32, path::Path, process, process::Stdio};
 
 #[derive(Debug, Parser)]
 #[clap(author, version, about, rename_all = "kebab-case")]
 pub struct Cli {
-    /// Run the command on the root workspace project.
-    #[clap(short, long)]
-    pub workspace_root: bool,
+    #[clap(flatten)]
+    pub config: Config,
+
     /// Command to execute.
     #[clap(subcommand)]
     pub command: Command,
 }
 
+/// Global configurations to be passed down to subcommands
+#[derive(Parser, Debug)]
+pub struct Config {
```

Comment:
Rename this to `GlobalOptions`.

Reply:
Done. renamed the struct to GlobalOptions and the Cli prop to global_options